src/test/kotlin/uk/gov/dluhc/printapi/service/PrintResponseBatchServiceTest.kt

```diff
@@ -0,0 +1,90 @@
+
+    @Mock
+    private lateinit var idFactory: IdFactory
+
+    @Test
+    fun `should update print details given successful batch responses`() {
+        // Given
+        val batchResponse1 = buildBatchResponse(status = SUCCESS)
+        val batchResponse2 = buildBatchResponse(status = SUCCESS)
+        val batchId1 = batchResponse1.batchId
+        val batchId2 = batchResponse2.batchId
+
+        val printDetails1 = listOf(buildPrintDetails(batchId = batchId1, status = SENT_TO_PRINT_PROVIDER))
+        val printDetails2 = listOf(buildPrintDetails(batchId = batchId2, status = SENT_TO_PRINT_PROVIDER))
+        given(repository.getAllByStatusAndBatchId(any(), any())).willReturn(printDetails1, printDetails2)
+
+        // When
+        service.processBatchResponses(listOf(batchResponse1, batchResponse2))
+
+        // Then
+        verify(repository).getAllByStatusAndBatchId(SENT_TO_PRINT_PROVIDER, batchResponse1.batchId)
+        verify(repository).getAllByStatusAndBatchId(SENT_TO_PRINT_PROVIDER, batchResponse2.batchId)
+        verify(repository).updateItems(printDetails1)
+        verify(repository).updateItems(printDetails2)
+        assertThat(printDetails1.plus(printDetails2)).extracting<Status> { it.status }
+            .allSatisfy { assertThat(it).isEqualTo(RECEIVED_BY_PRINT_PROVIDER) }
+        verifyNoInteractions(idFactory)
+    }
+
+    @Test
+    fun `should update print details given failed batch responses`() {
+        // Given
+        val batchResponse1 = buildBatchResponse(status = FAILED)
+        val batchResponse2 = buildBatchResponse(status = FAILED)
+        val batchId1 = batchResponse1.batchId
+        val batchId2 = batchResponse2.batchId
+
+        val printDetails1 = listOf(buildPrintDetails(batchId = batchId1, status = SENT_TO_PRINT_PROVIDER))
+        val printDetails2 = listOf(buildPrintDetails(batchId = batchId2, status = SENT_TO_PRINT_PROVIDER))
+        val newRequestId1 = aValidRequestId()
+        val newRequestId2 = aValidRequestId()
+        given(repository.getAllByStatusAndBatchId(any(), any())).willReturn(printDetails1, printDetails2)
+        given(idFactory.requestId()).willReturn(newRequestId1, newRequestId2)
+
+        // When
+        service.processBatchResponses(listOf(batchResponse1, batchResponse2))
+
+        // Then
+        verify(repository).getAllByStatusAndBatchId(SENT_TO_PRINT_PROVIDER, batchResponse1.batchId)
+        verify(repository).getAllByStatusAndBatchId(SENT_TO_PRINT_PROVIDER, batchResponse2.batchId)
+        verify(repository).updateItems(printDetails1)
+        verify(repository).updateItems(printDetails2)
+        verify(idFactory, times(2)).requestId()
+        assertThat(printDetails1.plus(printDetails2)).extracting<Status> { it.status }
+            .allSatisfy { assertThat(it).isEqualTo(PENDING_ASSIGNMENT_TO_BATCH) }
+        assertThat(printDetails1[0].requestId).isEqualTo(newRequestId1)
+        assertThat(printDetails2[0].requestId).isEqualTo(newRequestId2)
+    }
```

Comment:
Is it worth doing a test where one batch response is SUCCESS and one is FAILED ?

Reply:
Thanks, @NathanRussellValtech , for the suggestion.
I will change the tests accordingly.